src/ert/experiment_server/_server.py

```diff
@@ -141,12 +142,13 @@ def add_legacy_experiment(  # pylint: disable=too-many-arguments
             ensemble_size,
             current_case_name,
             args,
+            experiment_id,
```

Comment:
Why does it need to be set  and not within the constructor (or later on via the builder as we discussed) ?

Reply:
I'm not sure if I have strong preferences on this, so we can change it. But what we have today is builder for `job`, `step`, `stage`, `realization` and `ensemble`, but _not_ `experiment`.

We did discuss adding a builder for the experiment as well, with the counter argument being that we don't know up front what the ensemble will look like (successive ensembles are affected by the previous ones, with deactivated realizations). While when we create ensemble, we know for sure how the entire hierarchy will look like. Once a builder builds, the returned object should be completely populated, which we can't do for experiments as we only know what the first ensemble looks like. At the end the suggestions was to add an `EnsembleBuilder` instance to the `Experiment` object, with the at the time known information, which will be used to create ensembles. I believe we agreed to wait with this though.

For all the classes down the hierarchy today, they all accept an `id` in their `init`, so I suppose I did it this way to align with the rest. The builders all have a `set_id`, to be used when populating the instances themselves. Some builders accept that `id` is not set, and will then create them when trying to build. None of the classes create the `id` in their constructor.

In the end we have also discussed the point that the `id` could be the same as the one used in `storage`, but this is also implementing for the future, so not really a solid argument.